### PR TITLE
Add Linux command line option to show memory details

### PR DESCRIPTION
**Pull request by pyhys**
Show compile time options

### CODE REVIEW

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: sample_app/main_rtk.c [hunk 1]
```diff
@@ -121,28 +121,29 @@ static int _cmd_pnio_alarm_ack(
...
 {
-   unsigned long level = 0;
-   char *end = NULL;
+   unsigned long           level = 0x2010;  /*  See documentation for pnet_show()  */
+   char                    *end = NULL;
 
    if (argc == 2)
    {
       level = strtoul(argv[1], &end, 0);
    }
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/main_rtk.c:147]: Done

File: sample_app/main_rtk.c [hunk 2]
```diff
@@ -121,28 +121,40 @@ static int _cmd_pnio_alarm_ack(
...
+      printf("Usage: pnio_show level\n");
+      return -1;
+   }
+   else if (argc == 2)
    {
       level = strtoul(argv[1], &end, 0);
    }
+   if (level < 1)
+   {
+      printf("Error: The level must be an integer > 0\n");
```
Comment (fredrikdanielmoller) [inline sample_app/main_rtk.c:150]:
```suggestion
      shell_usage (argv[0], "level must be an integer > 0");
```

File: sample_app/main_rtk.c [hunk 3]
```diff
@@ -121,28 +121,40 @@ static int _cmd_pnio_alarm_ack(
...
+   if (argc > 2)
+   {
+      printf("Usage: pnio_show level\n");
+      return -1;
+   }
+   else if (argc == 2)
    {
       level = strtoul(argv[1], &end, 0);
    }
+   if (level < 1)
```
Comment (fredrikdanielmoller) [inline sample_app/main_rtk.c:148]:
```suggestion
   if (end == argv[1] || level == 0)
```

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: sample_app/main_rtk.c (hunk 3 above)
Comment (pyhys) [reply sample_app/main_rtk.c]: Changed to 
`if (level == 0)`
as we should allow the use of a default value.

File: sample_app/main_rtk.c (hunk 2 above)
Comment (pyhys) [reply sample_app/main_rtk.c]: Done

File: sample_app/main_rtk.c [hunk 4]
```diff
@@ -121,28 +121,40 @@ static int _cmd_pnio_alarm_ack(
...
 {
-   unsigned long level = 0;
-   char *end = NULL;
+   unsigned long           level = 0x2010;  /*  See documentation for pnet_show()  */
+   char                    *end = NULL;
 
-   if (argc == 2)
+   if (argc > 2)
+   {
+      printf("Usage: pnio_show level\n");
```
[the remark this answers is not in the document]
Comment (pyhys) [reply sample_app/main_rtk.c]: Done

File: sample_app/main_rtk.c [hunk 5]
```diff
@@ -121,28 +121,40 @@ static int _cmd_pnio_alarm_ack(
...
+   }
+   else if (argc == 2)
    {
       level = strtoul(argv[1], &end, 0);
    }
+   if (level == 0)
+   {
+      shell_usage (argv[0], "level must be an integer > 0");
+      return -1;
+   }
```
Comment (fredrikdanielmoller) [inline sample_app/main_rtk.c:152]:
```suggestion
   if (argc == 1)
   {
      level = 0x2010;  /*  See documentation for pnet_show()  */
   }
   else if (argc == 2)
   {
      level = strtoul(argv[1], &end, 0);
      if (end == argv[1] || level == 0)
      {
         shell_usage (argv[0], "level must be an integer > 0");
         return -1;
      }
   }
   else
   {
      shell_usage (argv[0], "wrong number of arguments");
      return -1;
   }
```

**Review by fredrikdanielmoller — CHANGES_REQUESTED** [review]

File: sample_app/main_rtk.c (hunk 5 above)
Comment (pyhys) [reply sample_app/main_rtk.c]: Thanks, done.

**Review by fredrikdanielmoller — APPROVED** [review]

**Pull request merged**